### PR TITLE
Handling cyclic dependencies for TypeScript federated execution

### PR DESCRIPTION
**Pull request by byeong-gil**
Relevant PR: https://github.com/lf-lang/reactor-ts/pull/198

This PR adds the handling of TPO levels, `PTAG`, and `port absent` messages for TS-targeted federated execution.

### CODE REVIEW

**Comment by byeong-gil** [comment]
I marked this PR as a "draft" again because I found non-determinism errors in this PR.

**Review by lhstrh — COMMENTED** [review]
Great to see that these test are now passing! Let's do another release of `reactor-ts` once https://github.com/lf-lang/reactor-ts/pull/198 is merged and then use `^0.6.0` in the `package.json`.

File: core/src/main/resources/lib/ts/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "name": "LinguaFrancaDefault",
     "type": "commonjs",
     "dependencies": {
-        "@lf-lang/reactor-ts": "^0.5.0",
+        "@lf-lang/reactor-ts": "git://github.com/lf-lang/reactor-ts.git#master",
```
Comment (lhstrh) [inline core/src/main/resources/lib/ts/package.json:5]:
```suggestion
        "@lf-lang/reactor-ts": "^0.6.0",
```

**Review by lhstrh — APPROVED** [review]

**Comment by byeong-gil** [comment]
ts-test (macos-latest) failed because a new conflict was emerged due to another auto-merging enabled branch.
Let me align this branch with the current `master` branch of `lingua-franca`.

**Review by hokeun — APPROVED** [review]

**Pull request merged**